### ISSUE TITLE
"section registered as allowDefinition='MachineToApplication' beyond application level" after creating new field type

### ISSUE THREAD

**Issue by bleroy** [bleroy/OrchardImportTest#499]
After creating a new field type and putting it in a new content type, the
moment I try to create a new item of the content type Visual Studio's debugger
stops the execution throwing the following exception:  
  
""It is an error to use a section registered as
allowDefinition='MachineToApplication' beyond application level. This error
can be caused by a virtual directory not being configured as an application in
IIS. (C:\Users\Cobo\Documents\Visual Studio 2010\Projects\thehoopsmarket-
development2\src\Orchard.Web\modules\thmcustomfields\web.config line 18)".  
  
I haven't modified the web.config file neither Orchard's or my project one.  
  
Software used:  

  * Windows 7 / IIS 7.5 
  * Visual Studio 2010 
  * SQL Server 2008 R2 Express 
  * Orchard 0.5   

Steps to reproduce:  

  1. Fresh installation running Orchard's latest release through VS, using a new blank SQL Server database: OK. 
  2. Adding a new MVC project under Orchard.Modules (files under src\Orchard.Web\modules): OK 
  3. Implemented the new field type: OK 
  4. Run Orchard, enabled both "Content Types" and my new module and type "ThmCustomFields" and "TrueFalseField": OK 
  5. Create a new content type from the administration menu including a field of my type: OK. 
  6. Create a new item of the just created content type: ERROR -> Above exception is thrown. 

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 16608](http://orchard.codeplex.com/workitem/16608)
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	sebastienros
**Reported on:**	Sep 5, 2010 at 7:12 PM
**Reported by:**	JacoboPolavieja
**Updated on:**	Apr 9, 2013 at 9:05 PM
**Updated by:**	stonstad
**Closed on:**	Nov 23, 2010 at 7:29 PM
**Closed by:**	sebastienros
**Status:**	Closed
**Reason Closed:**	Unassigned
**Assigned to:**	sebastienros
**Reported on:**	Sep 5, 2010 at 7:12 PM
**Reported by:**	JacoboPolavieja
**Updated on:**	Apr 9, 2013 at 9:05 PM
**Updated by:**	stonstad
**Closed on:**	Nov 23, 2010 at 7:29 PM
**Closed by:**	sebastienros

#### Binary Attachments

[ThmCustomFields.zip](http://orchard.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=orchard&WorkItemId=16608&FileAttachmentId=148474)

**Comment by bleroy** [comment]
On *2010-09-06 02:13:04 UTC*, **JacoboPolavieja** commented:

Sorry... just to clarify, I didn't use IIS 7.5, I run it through VS development server.

**Comment by bleroy** [comment]
On *2010-11-18 04:30:15 UTC*, **rpaquay** commented:

Sebastien, i think this might be "by design" of step 2. A module should not be created using MVC tooling, but using our codegen commands instead. Can you verify?

**Comment by bleroy** [comment]
On *2010-11-24 03:29:07 UTC*, **sebastienros** commented:

@Jacobo

You are trying to use a web.config which is not valid for embedding in a sub folder as it contains some sections which only aply at the application level. You should remove this file from your project.

**Comment by bleroy** [comment]
On *2010-11-24 05:39:06 UTC*, **JacoboPolavieja** commented:

If it is my mistake sorry for filing the bug then... Just to be clear, I did not touch anything on the original web.config nor I added any new web.config on the project. I just, as the steps say, run the downloaded copy of orchard as is. So, if I remove
 that file I wouldn't be able to even run Orchard if I'm not wrong, as there is just one web.config (the default one). Am I missing something?

Sorry for the inconveniences, thanks for everything.
